src/inmanta/execute/runtime.py

```diff
@@ -382,7 +382,8 @@ def receive_result(self, value: ListValue, location: Location) -> None:
     def listener(self, resultcollector: ResultCollector, location: Location) -> None:
         for value in self.value:
             resultcollector.receive_result(value, location)
-        self.listeners.append(resultcollector)
```

Comment:
this is a second bug triggered by the first one:

When the value is set, the listeners are dropped, causing this line to fail.

Reply:
The typing of `listeners` is very hard to follow but it doesn't look like it would be easy to fix. How comes listeners are added when the variable has already been frozen?

Reply:
I didn't dig into it very far, but: new statements being emitted (i.e. there is no reason why it should not happen, a statement can listen to something with a definite value, but then listeners must not be kept, the first half of the method feeds all values already.)